pkg/api/types.go

```diff
@@ -1007,6 +1007,23 @@ type PodSecurityContext struct {
 	// Use the host's ipc namespace.
 	// Optional: Default to false.
 	HostIPC bool `json:"hostIPC,omitempty"`
+
+	// Capabilities are the capabilities to add/drop when running containers
+	Capabilities *Capabilities `json:"capabilities,omitempty"`
+
+	// Run all containers in privileged mode
+	Privileged *bool `json:"privileged,omitempty"`
+
+	// SELinuxOptions is the SELinux context to be applied to all containers
+	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
+
+	// RunAsUser is the UID to run the entrypoint of all container processes
+	RunAsUser *int64 `json:"runAsUser,omitempty"`
+
+	// RunAsNonRoot indicates that all containers should be run as a non-root user.
+	// If the RunAsUser field is not explicitly set then the kubelet may check the
+	// image for a specified user or perform defaulting to specify a user.
+	RunAsNonRoot bool
```

Comment:
Wow, we do.  It's missing in master.

On Thu, Oct 8, 2015 at 11:17 AM, Sami Wagiaalla <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/14705#discussion_r41526648>
> :
>
> > +	// Capabilities are the capabilities to add/drop when running containers
> > +	Capabilities *Capabilities `json:"capabilities,omitempty"`
> > +
> > +	// Run all containers in privileged mode
> > +	Privileged *bool `json:"privileged,omitempty"`
> > +
> > +	// SELinuxOptions is the SELinux context to be applied to all containers
> > +	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
> > +
> > +	// RunAsUser is the UID to run the entrypoint of all container processes
> > +	RunAsUser *int64 `json:"runAsUser,omitempty"`
> > +
> > +	// RunAsNonRoot indicates that all containers should be run as a non-root user.
> > +	// If the RunAsUser field is not explicitly set then the kubelet may check the
> > +	// image for a specified user or perform defaulting to specify a user.
> > +	RunAsNonRoot bool
>
> Do we need json:"runAsNonRoot,omitempty" here ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14705/files#r41526648>.
>